packages/graphql_server/src/api/githubApi.ts

```diff
@@ -65,3 +71,71 @@ export async function getUserInfo(query: string, authToken: string): Promise<Git
   )
   return response.data.data.user
 }
+
+/**
+ * Returns a Array of Founders with their names, login names and twitter handles. This method goes trough the commit history of a specific repo
+ * and fetches teh first 5 commits, which are most likley the initiators of a project. It then removes duplicates, because several commits can be from the
+ * same person, but shouldn't be returned within the Array
+ * @param owner: name of the owner of the github repo
+ * @param name: name of the github repo
+ * @returns An Array of the project founders
+ */
+export async function getRepoFounders(owner: string, name: string): Promise<ProjectFounder[]> {
+  if (!owner || !name) {
+    throw new Error('Not able to fetch repository to get founders of the project')
+  }
+
+  const query = `query {
+        repository(owner: "${owner}", name: "${name}") {
+          defaultBranchRef {
+            target {
+              ... on Commit {
+                history(first: 5) {
+                  edges {
+                    node {
+                      author {
+                        user {
+                          name
+                          login
+                          twitterUsername
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }`
+
+  // Building the AxiosRequest and initiate the post request
+  const authToken = 'Bearer ' + (process.env.GITHUB_API_TOKEN || '')
+  const response: AxiosResponse<{ data: { repository: GitHubCommitHistory } }> = await axios.post(
+    'https://api.github.com/graphql',
+    {
+      query
+    },
+    {
+      headers: {
+        Authorization: authToken
+      }
+    }
+  )
+
+  const distinctCommiters: ProjectFounder[] = []
+
+  // checks, whether a login name appears twice and only pushes distinct founders into the array
+  response.data.data.repository.defaultBranchRef.target.history.edges.forEach((node) => {
```

Comment:
response?.data?.data everywhere, also check for existing edges or throw error, also make sure to use optional chaining below